cloudsql/mysql/database-sql/cloudsql.go

```diff
@@ -156,15 +156,11 @@ func mustConnect() *sql.DB {
 
 	// Use the connector when INSTANCE_CONNECTION_NAME (proj:region:instance) is defined.
 	if os.Getenv("INSTANCE_CONNECTION_NAME") != "" {
-		if os.Getenv("DB_USER") == "" && os.Getenv("DB_IAM_USER") == "" {
-			log.Fatal("Warning: One of DB_USER or DB_IAM_USER must be defined")
-		}
-		// DB_IAM_USER takes precedence if both environment variables are
-		// defined.
-		if os.Getenv("DB_IAM_USER") != "" {
-			db, err = connectWithConnectorIAMAuthN()
-		} else {
+		// Use IAM Authentication (recommended) if password is not set
+		if os.Getenv("DB_PASS") != "" {
```

Comment:
I worry a bit about users forgetting to set a password and then getting an error about IAM Authn. It's the difference between an implicit and explicit API.

Reply:
I think that's a fair perspective. The reason I changed it was because I had the opposite experience getting started. I thought I needed a password even when using IAM Authentication.

When I was trying to get this repository working with IAM, I had to read through the code to realize that I needed to change the environment variable. I also spent a while thinking, "Even though this is an IAM user, maybe I still need to include the database instance password?"

That seems totally backwards now, but it took me a while to figure out.

Reply:
Per your recommendation, I added it back.

I was able to update `deployment.yaml` to the point where I think I'm happy with the result, and I think having a separate `DB_IAM_USER` allows us to clearly split out which environment variable controls IAM and which variable controls the method of connection. Hope it looks good!